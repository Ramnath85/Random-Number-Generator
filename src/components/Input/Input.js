import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ onClick, onChange }) => (
  <div>
    <input
      type="number"
      placeholder="Enter a number"
      onChange={onChange}
    />
    <button onClick={onClick}>Generate Phone Number(s)</button>
  </div>
);

Input.propTypes = {
  onClick: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Input;