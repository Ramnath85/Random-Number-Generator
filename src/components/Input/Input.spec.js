import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from "./Input";

configure({ adapter: new Adapter() });

describe('Test Input Component', () => {
  let shallowComponent;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      onClick: () => {},
      onChange: () => {}
    };
    shallowComponent = shallow(<Input {...defaultProps} />);
  });

  test('should render the Input component', () => {
    expect(shallowComponent).toMatchSnapshot();
    expect(shallowComponent.length).toEqual(1)
  });
});