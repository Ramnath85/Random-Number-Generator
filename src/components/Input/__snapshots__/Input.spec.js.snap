// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component should render the Input component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    onChange={[Function]}
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Enter a number"
          type="number"
        />,
        <button
          onClick={[Function]}
        >
          Generate Phone Number(s)
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Enter a number",
          "type": "number",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Generate Phone Number(s)",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Generate Phone Number(s)",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Enter a number"
            type="number"
          />,
          <button
            onClick={[Function]}
          >
            Generate Phone Number(s)
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Enter a number",
            "type": "number",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Generate Phone Number(s)",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Generate Phone Number(s)",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
